src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ModuleToken.cs

```diff
@@ -64,16 +65,12 @@ public bool Equals(ModuleToken moduleToken)
             return Module == moduleToken.Module && Token == moduleToken.Token;
         }
 
-        public int CompareTo(ModuleToken other)
+        public int CompareTo(ModuleToken other, TypeSystemComparer comparer)
         {
-            int result = 0;
-            // TODO: how to compare modules?
-            // result = Module.CompareTo(other.Module);
-            if (result == 0)
-            {
-                result = Token.CompareTo(other.Token);
-            }
-            return result;
+            int result = Module.CompareTo(other.Module);
+            if (result != 0) return result;
+
+            return Token.CompareTo(other.Token);
```

Comment:
Can we compare the token first? Might be a bit faster to compare a uint before digging deeper :)

Reply:
Yes, good idea.